src/pages/index.js

```diff
@@ -1,100 +1,259 @@
+    saveProfileButton.textContent = "Saving...";
+
+    // Send PATCH request to update the user's profile info
+    api
+      .updateUserInfo({
+        name: formData.name,
+        about: formData.description,
+      })
+      .then((updatedUserInfo) => {
+        userInfoInstance.setUserInfo({
+          name: updatedUserInfo.name,
+          description: updatedUserInfo.about,
+          avatar: updatedUserInfo.avatar,
+        });
+        profileEditPopupInstance.close(); // Close the popup after success
+      })
+      .catch((err) => {
+        console.error("Error updating profile:", err); // Handle errors during profile update
+      })
+      .finally(() => {
+        saveProfileButton.textContent = "Save"; // Revert the button text
+      });
   }
 );
-profileEditPopupInstance.setEventListeners(); // Set up event listeners for the popup
 
-// Initialize the add card popup with form handling
+profileEditPopupInstance.setEventListeners(); // Set event listeners for the profile edit popup
+
+// Initialize the card addition popup with form submission handling
 const addCardPopupInstance = new PopupWithForm(
   "#add-card-popup",
   (formData) => {
-    const cardElement = createCard({
-      name: formData.title, // Title input from the form
-      link: formData.url, // URL input from the form
-    });
-    cardFormValidator.disableButton();
-    cardSection.addItem(cardElement); // Add the new card to the card section
-    addCardPopupInstance.close(); // Close the popup after submission
+    saveCardButton.textContent = "Saving...";
+
+    // Send POST request to add a new card
+    api
+      .addCard({
+        name: formData.title, // Card title
+        link: formData.url, // Card image URL
+      })
+      .then((newCard) => {
+        const cardElement = createCard({
+          name: newCard.name,
+          link: newCard.link,
+          _id: newCard._id,
+          owner: newCard.owner,
+          likes: newCard.likes,
+        });
+        cardSection.addItem(cardElement); // Add new card to the UI
+        addCardPopupInstance._formElement.reset(); // Reset form fields after submission
+        addCardPopupInstance.close(); // Close the popup after success
+        saveCardButton.textContent = "Save"; // Reset button text
+      })
+      .catch((err) => console.error("Error adding card:", err)); // Handle errors during card addition
   }
 );
-addCardPopupInstance.setEventListeners(); // Set up event listeners for the popup
 
-// Initialize the image popup for displaying larger images
+addCardPopupInstance.setEventListeners(); // Set event listeners for the card addition popup
+
+// Initialize avatar edit popup with form submission handling
+const avatarEditPopup = new PopupWithForm("#avatar-edit-popup", (formData) => {
+  saveAvatarButton.textContent = "Saving...";
+
+  // Check if the form is valid before sending the API request
+  if (!avatarFormValidator._hasInvalidInput()) {
```

Comment:
`  if (!avatarFormValidator._hasInvalidInput()) {
`

It's not neede because `validation` handles everything inside `FormValidator`